substra/sdk/schemas.py

```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
     def is_many(self):
         return self.paths and len(self.paths) > 0
 
+    @pydantic.field_validator("paths")
+    @classmethod
+    def resolve_paths(cls, v: typing.Any) -> typing.Any:
+        """Resolve given paths."""
+        if v is None:
+            raise ValueError("'paths' cannot be set to None.")
+
+        paths = []
+        for path in v:
+            path = pathlib.Path(path)
```

Comment:
```suggestion
```

pydantic already casts str to path when using the type `pathlib.Path`

Reply:
Oh I didn't knew ! So nice...